src/org/zaproxy/zap/authentication/PostBasedAuthenticationMethodType.java

```diff
@@ -296,7 +343,11 @@ public WebSession authenticate(SessionManagementMethod sessionManagementMethod,
 
 			// Send the authentication message
 			try {
-				getHttpSender().sendAndReceive(msg);
+				if(this.requireAnitCsrfTokenWhenLogin) {
```

Comment:
Why?

Reply:
Well, DVWA redirects to index page when successfully authenticated. I assumed this is the general case. With the redirect response header I do not get anything useful to mark as a logged-in/logged-out indicator. 
![image](https://user-images.githubusercontent.com/17596394/40310646-167eda8c-5d2b-11e8-9b0b-2a96d6a7efd0.png)

I marked "You have logged in as 'admin'" in index page as logged-in indicator. Without redirection as the logged in indicator is not present in the response, the authentication is marked as failed.

Reply:
That should also be part of the logged in indicator (e.g. `Location: index.php`), if not other parts of ZAP will fail to know that the message is authenticated (not everything follows redirections automatically).

Reply:
Initially, I went with `Location: index.php`. But then I found the logged in/ logged out indicator must be something that is common among all pages. Correct me if I am wrong. As each response goes through `isAuthenticated` process(?) where the response header and body are checked for any match of the `loggedInIndicator` or `loggedOutIndicator` value, this value must be something that presents in all pages.
Though I marked in index.php the logout link and logged in message appear on all pages. That's why I moved from `Location: index.php`. Am I missing anything?

Reply:
The logged in/out indicators can be composed to include both cases, e.g.:
```
(Location: index.php)|(You have logged in as)
```